checker/src/constant_domain.rs

```diff
@@ -770,6 +772,27 @@ impl ConstantValueCache {
         })
     }
 
+    /// Does an expensive check to see if the given function is mirai_annotations.mirai_precondition.
+    /// Once it finds the function it caches it so that subsequent checks are cheaper.
+    pub fn check_if_mirai_precondition_function(&mut self, fun: &ConstantDomain) -> bool {
```

Comment:
Ah, this cached summary key pattern shows up multiple times :) Can you abstract the code below? It's a bit complex, and if it's going to be repeated then it'd be nice to have some code sharing.

Reply:
#125